db/pgstorage/migrations/0009.sql

```diff
@@ -0,0 +1,15 @@
+-- +migrate Up
+
+ALTER TABLE sync.deposit
+ADD COLUMN IF NOT EXISTS origin_rollup_id BIGINT DEFAULT 0;
+
+ALTER TABLE sync.claim DROP CONSTRAINT claim_pkey;
+ALTER TABLE sync.claim ADD PRIMARY KEY (index, rollup_index, mainnet_flag);
```

Comment:
PK used to be `index, metwork_id`. This is wrong because `network_id` references destination network. Therefore there can be many claims with the same index (deposit count) and destination network. Example:
- Rollup 1 does the first deposit to L1
- Rollup 2 does the first deposit to L1
Once both deposits are claimed on L1, they will same index and and network id.

OTOH, index is unique per origin network, it's a counter (deposit count). And the way to index the origin network is with rollup index and mainnet flag, as explained in this comment:


```
// origin rollup ID is calculated as follows:
	// // if mainnet_flag: 0
	// // else: rollup_index + 1
	// destination rollup ID == network_id: network that has received the claim, therefore, the destination rollupID of the claim
```